app/app/src/main/kotlin/com/hedvig/app/feature/profile/ui/payment/PaymentDestination.kt

```diff
@@ -196,30 +239,36 @@ private fun AddDiscount(uiState: PaymentUiState) {
     Spacer(Modifier.height(4.dp))
   }
   Spacer(Modifier.height(12.dp))
-  Row(
-    modifier = Modifier.fillMaxWidth(),
-    horizontalArrangement = Arrangement.SpaceBetween,
+  AnimatedVisibility(
+    visible = showDiscountInput,
+    enter = fadeIn(),
+    exit = fadeOut(),
   ) {
-    HedvigTextField(
-      value = uiState.discountCode ?: "",
-      onValueChange = {},
-      errorText = uiState.discountError,
-      label = {
-        Text(stringResource(id = R.string.REFERRAL_ADDCOUPON_INPUTPLACEHOLDER))
-      },
-      modifier = Modifier.weight(1f),
-    )
-    Spacer(Modifier.width(8.dp))
-    HedvigContainedSmallButton(
-      text = stringResource(id = R.string.PAYMENTS_ADD_CODE_BUTTON_LABEL),
-      onClick = {},
-      contentPadding = PaddingValues(horizontal = 16.dp, vertical = 16.dp),
-      colors = ButtonDefaults.buttonColors(
-        containerColor = MaterialTheme.colorScheme.typeContainer,
-        contentColor = MaterialTheme.colorScheme.primary,
-      ),
-      modifier = Modifier.widthIn(min = 127.dp),
-    )
+    Row(
+      modifier = Modifier.fillMaxWidth(),
+      horizontalArrangement = Arrangement.SpaceBetween,
+    ) {
+      HedvigTextField(
+        value = uiState.discountCode?.code ?: "",
+        onValueChange = { onDiscountCodeChanged(CampaignCode(it)) },
+        errorText = uiState.discountError,
+        label = {
+          Text(stringResource(id = R.string.REFERRAL_ADDCOUPON_INPUTPLACEHOLDER))
+        },
+        modifier = Modifier.weight(1f),
+      )
+      Spacer(Modifier.width(8.dp))
+      HedvigContainedSmallButton(
+        text = stringResource(id = R.string.PAYMENTS_ADD_CODE_BUTTON_LABEL),
+        onClick = onAddDiscountCode,
+        contentPadding = PaddingValues(horizontal = 16.dp, vertical = 16.dp),
+        colors = ButtonDefaults.buttonColors(
+          containerColor = MaterialTheme.colorScheme.typeContainer,
+          contentColor = MaterialTheme.colorScheme.primary,
+        ),
+        modifier = Modifier.widthIn(min = 127.dp),
+      )
+    }
```

Comment:
I think what we want in this case for the small button is for it to be exactly as big as the text field on the left. And then for it to take as much space as it needs, and the rest can be give to the TextField to the left
```suggestion
    Row(
      modifier = Modifier.fillMaxWidth().height(IntrinsicSize.Max),
      horizontalArrangement = Arrangement.SpaceBetween,
    ) {
      HedvigTextField(
        value = uiState.discountCode?.code ?: "",
        onValueChange = { onDiscountCodeChanged(CampaignCode(it)) },
        errorText = uiState.discountError,
        label = {
          Text(stringResource(id = R.string.REFERRAL_ADDCOUPON_INPUTPLACEHOLDER))
        },
        modifier = Modifier.weight(1f),
      )
      Spacer(Modifier.width(8.dp))
      HedvigContainedSmallButton(
        text = stringResource(id = R.string.PAYMENTS_ADD_CODE_BUTTON_LABEL),
        onClick = onAddDiscountCode,
        contentPadding = PaddingValues(horizontal = 32.dp, vertical = 16.dp),
        colors = ButtonDefaults.buttonColors(
          containerColor = MaterialTheme.colorScheme.typeContainer,
          contentColor = MaterialTheme.colorScheme.primary,
        ),
        modifier = Modifier.fillMaxHeight(),
      )
    }
```

Would this work instead?
This way we make the row use IntrinsicSize.Max, so that it finds which one of the children is bigger in height (the text field for us) and then it also measures that high. This allows for the button to then be able to do `fillMaxHeight()` and that will make it be as big as that row, which in turn is as big as the TextField.

And then by adjusting the contentPadding to have 32.dp horizontal padding we get the result that design seems to want from this specific button.

Now if this makes it so that the button takes way too much space on the right side, leaving nothing to the input, maybe we can just put the horizontal padding back to 16, but still not have the hardcoded 127 width as that'll probably make it look weird depending on the localization and how big the inner text is.